### PR TITLE
Integrate notifier with drivers

### PR DESCRIPTION
**Pull request by hackaugusto**
~Merge/Review after https://github.com/aiven/rohmu/pull/66~

Add support for the notifier to each driver

### CODE REVIEW

File: rohmu/__init__.py [hunk 1]
```diff
@@ -44,8 +46,29 @@ def get_class_for_transfer(obj_store: Config) -> Type[BaseTransfer]:
...
+def get_class_for_notifier(notifier_config: dict) -> Type[Notifier]:
+    notifier_type = notifier_config[NOTIFIER_TYPE]
+    if notifier_type == "http":
+        from .notifier.http import BackgroundHTTPNotifier
+
+        return BackgroundHTTPNotifier
+    raise InvalidConfigurationError("unsupported storage type {0!r}".format(notifier_type))
+
+
+def get_notifier(notifier_config: dict) -> Notifier:
```
Comment (ngilles-aiven) [inline rohmu/__init__.py:58]: Do we want  `mypy` to complain on bare generics?

File: rohmu/object_storage/azure.py [hunk 2]
```diff
@@ -199,6 +214,7 @@ def delete_key(self, key):
             return blob_client.delete_blob()
         except azure.core.exceptions.ResourceNotFoundError as ex:  # pylint: disable=no-member
             raise FileNotFoundFromStorageError(path) from ex
+        self.notifier.object_deleted(key)
```
Comment (ngilles-aiven) [inline rohmu/object_storage/azure.py:217]: This would not get called, as the `return` is in `try:` block

File: rohmu/object_storage/swift.py [hunk 3]
```diff
@@ -249,9 +252,9 @@ def store_file_from_memory(self, key, memstring, metadata=None, cache_control=No
 
         path = self.format_key_for_backend(key)
         metadata_to_send = self._metadata_to_headers(self.sanitize_metadata(metadata))
-        self.conn.put_object(
-            self.container_name, path, contents=bytes(memstring), content_type=mimetype, headers=metadata_to_send
-        )
+        data = bytes(memstring)
+        self.conn.put_object(self.container_name, path, contents=data, content_type=mimetype, headers=metadata_to_send)
+        self.notifier.object_deleted(key=key, size=len(data))
```
Comment (ngilles-aiven) [inline rohmu/object_storage/swift.py:257]: should be `object_created`

File: rohmu/object_storage/swift.py [hunk 4]
```diff
@@ -265,6 +268,7 @@ def store_file_from_disk(self, key, filepath, metadata=None, multipart=None, cac
                 mimetype=mimetype,
                 content_length=obsz,
             )
+            self.notifier.object_deleted(key=key, size=os.path.getsize(fp))
```
Comment (ngilles-aiven) [inline rohmu/object_storage/swift.py:271]: should be `object_created`

**Review by ngilles-aiven — CHANGES_REQUESTED** [review]
Mostly LGTM, but a couple of what look like copy/paste errors, and a misplaced call.

File: rohmu/object_storage/azure.py (hunk 2 above)
Comment (hackaugusto) [reply rohmu/object_storage/azure.py:217]: oops, good caught :)

File: rohmu/__init__.py (hunk 1 above)
Comment (hackaugusto) [reply rohmu/__init__.py:58]: what do you mean? `Notifier` should be an `ABC` not a `TypeVar`.

Comment (ngilles-aiven) [reply rohmu/__init__.py:58]: Sorry, I meant for the `dict`, in my brain I had specified it :-D

Comment (hackaugusto) [reply rohmu/__init__.py:58]: AH, I tried to do a best effort typing on this PR. So I wasn't really trying to chase what are the correct types for the values. What do you suggest?

Comment (ngilles-aiven) [reply rohmu/__init__.py:58]: I don't suggest anything :-p I think it's just I'm used to `mypy` moaning about this in other projects I guess 😓

**Review by ngilles-aiven — APPROVED** [review]
LGTM

**Pull request merged**